addons/account/wizard/account_payment_register.py

```diff
@@ -564,17 +580,27 @@ def _get_total_amount_in_wizard_currency_to_full_reconcile(self, batch_result, e
                 self.payment_date,
             ), False
 
+    @api.onchange('amount')
+    def _onchange_amount(self):
+        if self.amount_change_mode != 'code_change':
+            self.is_user_change_amount = True
+        else:
+            self.is_user_change_amount = False
+
     @api.depends('can_edit_wizard', 'source_amount', 'source_amount_currency', 'source_currency_id', 'company_id', 'currency_id', 'payment_date')
     def _compute_amount(self):
         for wizard in self:
-            if not wizard.journal_id or not wizard.currency_id or not wizard.payment_date:
-                wizard.amount = wizard.amount
-            elif wizard.source_currency_id and wizard.can_edit_wizard:
-                batch_result = wizard._get_batches()[0]
-                wizard.amount = wizard._get_total_amount_in_wizard_currency_to_full_reconcile(batch_result)[0]
-            else:
-                # The wizard is not editable so no partial payment allowed and then, 'amount' is not used.
-                wizard.amount = None
+            if wizard.old_payment_date == wizard.payment_date or not wizard.is_user_change_amount:
```

Comment:
`old_payment_date` is the core of it, How I will know that `_compute_amount` triggered because of changing in the `payment_date`? 

The other two fields are used to differentiate between who has changed the amount 'user' or 'code'.

Reply:
Done